tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11236,1074 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+  LongVectorOpTestBase<int64_t>(LongVectorOpType_Clamp);
+}
+
+TEST_F(ExecutionTest, LongVector_Initialize_int64) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<int64_t>(LongVectorOpType_Initialize);
+}
+
+TEST_F(ExecutionTest, LongVector_Clamp_uint16) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint16_t>(LongVectorOpType_Clamp);
+}
+
+TEST_F(ExecutionTest, LongVector_Initialize_uint16) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint16_t>(LongVectorOpType_Initialize);
+}
+
+TEST_F(ExecutionTest, LongVector_Clamp_uint32) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint32_t>(LongVectorOpType_Clamp);
+}
+
+TEST_F(ExecutionTest, LongVector_Initialize_uint32) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint32_t>(LongVectorOpType_Initialize);
+}
+
+TEST_F(ExecutionTest, LongVector_Clamp_uint64) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint64_t>(LongVectorOpType_Clamp);
+}
+
+TEST_F(ExecutionTest, LongVector_Initialize_uint64) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorOpTestBase<uint64_t>(LongVectorOpType_Initialize);
+}
+
+template <typename T>
+void ExecutionTest::LongVectorOpTestBase(LongVectorOpType opType) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  LongVectorOpTestConfig<T> TestConfig(opType);
+
+  LongVectorOpTestBase<T, 3>(TestConfig);
+  LongVectorOpTestBase<T, 4>(TestConfig);
+  LongVectorOpTestBase<T, 5>(TestConfig);
+  LongVectorOpTestBase<T, 16>(TestConfig);
+  LongVectorOpTestBase<T, 17>(TestConfig);
+  LongVectorOpTestBase<T, 35>(TestConfig);
+  // TODOLONGVEC: 100, 1024 breaks the size limit for structured buffers
+  // LongVectorOpTestBase<T, 100>(TestConfig);
+  // LongVectorOpTestBase<T, 1024>(TestConfig);
+}
+
+template <typename T, std::size_t N>
+void ExecutionTest::LongVectorOpTestBase(
+    LongVectorOpTestConfig<T> &TestConfig) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  LogCommentFmt(L"Running LongVectorOpTestBase<%S, %zu>", typeid(T).name(), N);
+
+  CComPtr<ID3D12Device> D3DDevice;
+  if (!CreateDevice(&D3DDevice, D3D_SHADER_MODEL_6_9)
+      && !m_ExperimentalModeEnabled) {
+    if(m_HLKModeEnabled)
```

Comment:
This seems unintentional, but 12119-12122 is dead code because the return isn't under scope in 12115

Reply:
Yah, that was a mistake. Fixed.